### PR TITLE
Compile error fixes + negative filter patterns + total_time metric for httpunit collector

### PR DESCRIPTION
**Pull request by rajder**
Can now disable collectors using filter patterns prefixed by "-", eg. scollector-windows-amd64.exe -p -d -l -f="network_windows,-tcp" on my dev. machine prints:
bosun.org/cmd/scollector/collectors.c_network_windows

(excluding c_network_windows_tcp)

I needed this specifically for disabling c_network_windows_tcp while keeping c_network_windows enabled, trying to find performance issues caused by one of the two collectors, or both. High CPU usage by WmiPrvSE.exe and blocked collectors seems to be the symptoms (have not solved the issue or confirmed the offending collector yet)
Edit: It was not one of the network collectors. Could be some WMI related issue. It seems like the issue mostly occurs when the machines are under heavy load.

### CODE REVIEW

**Comment by rajder** [comment]
Implemented another feature in this branch since I need the previous changes I made. BR /AR

**Comment by lemkepf** [comment]
the negative filter patterns would be awesome. Example: I want to disable cassandra monitoring, but everything else should stay on.

File: _third_party/github.com/StackExchange/httpunit/httpunit.go [hunk 1]
```diff
@@ -393,6 +393,7 @@ type TestResult struct {
 	GotText     bool
 	GotRegex    bool
 	InvalidCert bool
+	TimeTotal   time.Duration
```
Comment (captncraig) [inline _third_party/github.com/StackExchange/httpunit/httpunit.go]: You shouldn't edit files in _third_party directly. Instead this feature will need to be a seperate pr to github.com/StackExchange/httpunit

File: build/build.go [hunk 2]
```diff
@@ -37,7 +37,7 @@ func main() {
 	}
 
 	timeStr := time.Now().UTC().Format("20060102150405")
-	ldFlags := fmt.Sprintf("-X bosun.org/version.VersionSHA %s -X bosun.org/version.VersionDate %s", sha, timeStr)
```
Comment (captncraig) [inline build/build.go]: What error is this fixing? This code successfully sets the appropriate variables in all of our builds. Is it not working for you?

Comment (rajder) [inline build/build.go]: The Go 1.5 compiler prints warnings about the old syntax.

File: _third_party/github.com/StackExchange/httpunit/httpunit.go (hunk 1 above)
Comment (rajder) [inline _third_party/github.com/StackExchange/httpunit/httpunit.go]: Ok I will look into that next week.

Comment (rajder) [inline _third_party/github.com/StackExchange/httpunit/httpunit.go]: I merged over my changes to httpunit, I did not merge back the changes from httpunit here yet (some timout improvements): https://github.com/StackExchange/httpunit/pull/5

**Comment by rajder** [comment]
Yet another small feature in this fork, Freq setting for HTTPUnit collectors. The default Freq if missing in the toml file is the same as the previously hardcoded value, 5 minutes.

**Comment by captncraig** [comment]
Sorry this has sat so long. Can you please squash and rebase to master?

File: cmd/scollector/conf/conf.go [hunk 3]
```diff
@@ -119,6 +119,7 @@ type ProcessDotNet struct {
 type HTTPUnit struct {
 	TOML  string
 	Hiera string
+	Freq  int
```
Comment (captncraig) [inline cmd/scollector/conf/conf.go]: I'd like this documented. Is this seconds? minutes? something else? cmd/scollector/doc.go should have docs for this.

File: build.scollector_win.bat [hunk 4]
```diff
@@ -0,0 +1,33 @@
+@echo off
+rem The purpose of this script is to make building on windows easier.
```
Comment (captncraig) [inline build.scollector_win.bat]: What specifically does this do that build.go does not?

**Comment by rajder** [comment]
Squash rebase, sure np. Should I remove the windows build helper script? It's just a personal convenience thing to enable autocomplete from windows command line.

**Comment by gbrayut** [comment]
I would say remove the build.scollector_win.bat file, as our build scripts are usually based in go. You can keep it in your local repo and add it to a [global ignore](https://help.github.com/articles/ignoring-files/) if you still want to use it.

**Comment by rajder** [comment]
Rebased, squashed, added some info for Freq in cmd/scollector/doc.go.

**Comment by gbrayut** [comment]
Sorry for the delay. This was merged as part of #1630

**Comment by rajder** [comment]
Oh, cool. Thanks for the info.

**Pull request closed without merging**